install_and_upgrade.sh

```diff
@@ -106,6 +106,22 @@ fi
 # source some env vars
 . "$HOME/.noaa-v2.conf"
 
+log_running "Installing certbot for SSL certificates signed by the Let's Encrypt..."
+if [ $? -eq 0 ]; then
+  sudo apt install certbot
+else
+  die "  Something failed with the install - please inspect the logs above"
+fi
+
+log_running "Installing SSL certificates..."
```

Comment:
I would rather this was a separate contribution to master and not conflated into this branch and PR.

That being said, I am a little concerned that doing this "by default" encourages people to open this to the web (to get a cert) which I am not thrilled about. Also I think that we need to make this auto-renew (this assumes people run this script every three months), finally, I think this would be better done in ansible than in this script (it would be easier to configure the cron, fall back to self-signed if you can't get a real cert etc)